### DISCUSSION TITLE
Difficulties testing vega branch using Dev Container/Codespaces

### DISCUSSION THREAD

**Discussion by Mark-LaCroix** [requarks/wiki#6611] [Help / Questions]
I've been trying to test out 3.0/vega following the instructions in the README, with Docker/VSCode Dev Container locally (and also on a GitHub Codespace) but I can't get it working. Following the instructions, I get stuck at `node run start` where it reports that it's missing a "run" module.

![image](https://github.com/requarks/wiki/assets/1129348/6cb7be94-ef83-4fef-a6c5-0f918bee2453)

If I back out to the root folder and use `node server` instead, it does start up, but halts complaining about a missing favicon.ico file.

If I add that file and try again, the server _will_ start, but when I navigate to it in a browser an error about a different missing node module ("pug") is printed to the screen:

![image](https://github.com/requarks/wiki/assets/1129348/a185ddd4-77c3-4edf-b561-aef9422736f3)

So, if I install pug (`npm install pug` from the project's root folder) and try to `node server` again, I get this odd message "Must run server from the parent directory!" which I can't make heads or tails of:

![image](https://github.com/requarks/wiki/assets/1129348/a43aab01-82a1-4425-9112-29773fe37c20)

Obviously things with this branch are in flux, so should I just sit tight and wait for future updates to the vega branch before trying again, or am I doing something wrong here?

**Comment by NGPixel** [comment]
The command is wrong. It should say `npm run server`, not `node run start`.
Also, you seem to be missing npm packages. Make sure to run `npm install` first in both `server/` and `ux/`.

You shouldn't run any npm command from the root folder, nor install npm package, nor create missing files.

**Reply by Mark-LaCroix** [reply]
> Make sure to run `npm install` first in both `server/` and `ux/`.

Sure, I was just following the instructions to the letter. As it happens, running it in `server/` doesn't install any new modules, but in `ux/` it installs _many_, which is ironic because `/ux` didn't give me any problems.

> The command is wrong. It should say `npm run server`, not `node run start`.

Ah, if I used node more often I might have noticed that error in the README myself. _But_ the line should actually be `npm run start`.

Either way, running the correct command in `/server` doesn't change anything. I still get the missing favicon file, the missing "pug" module, etc., same as before. This makes sense, I guess, as there's not really any difference between `/workspace $ node server` (what I was doing) and `/workspace/server $ npm run start` (what I'm doing now).

If I try to fix things as before (installing pug in `/server`), I get past it, but then this warning:
```
Error: Failed to lookup view "error" in views directory "/workspace/server/views"
    at Function.render (/workspace/server/node_modules/express/lib/application.js:597:17)
    **[... trace continues ...]**
    at process.processImmediate (node:internal/timers:478:21)
```

(I'd bet that's not very helpful to include, but just in case...)

> You shouldn't run any npm command from the root folder, nor install npm package, nor create missing files.

Obviously it didn't help, I was just futzing around trying to figure out if I could get it working.

I'm no closer to a solution now, either because I don't know enough about node to troubleshoot, there's something wrong/missing with the container, or some other reason I'm not qualified to diagnose. But I am following the setup precisely.

In the end, I just wanted to try it out and see what state it was in, but I don't really have time to debug further than this, and it's probably not worth your time to help me if it's something on my end, so I will give up on it for now. Still looking forward to the eventual release, though.

If anyone else has had luck, let me know.
